src/Classes/MonsterHunt.uc

```diff
@@ -424,17 +426,35 @@ function ScoreKill(pawn Killer, pawn Other) {
 // =========================================================================
 // Score depending on which monster type the player kills
 
-	if (Killer.bIsPlayer) {
+	if (Killer.bIsPlayer && Killer.PlayerReplicationInfo != None) {
 		if (Other.IsA('Titan') || Other.IsA('Queen') || Other.IsA('WarLord')) Killer.PlayerReplicationInfo.Score += 4;
 		else if (Other.IsA('GiantGasBag') || Other.IsA('GiantManta')) Killer.PlayerReplicationInfo.Score += 3;
 		else if (Other.IsA('SkaarjWarrior') || Other.IsA('MercenaryElite') || Other.IsA('Brute')) Killer.PlayerReplicationInfo.Score += 2;
 		// Lose points for killing innocent creatures. Shame ;-)
-		else if (Other.IsA('Nali') || Other.IsA('Cow') || Other.IsA('NaliRabbit')) Killer.PlayerReplicationInfo.Score -= 6;
+		else 	if ( Other.IsA('Nali') || Other.IsA('Cow') )
+		{
+			switch (ScriptedPawn(Other).AttitudeToCreature(Killer))
+			{
+				case ATTITUDE_Hate:
+				case ATTITUDE_Frenzy:
+					bEnemy = true;
+			}
+			switch (Other.AttitudeToPlayer)
+			{
+				case ATTITUDE_Hate:
+				case ATTITUDE_Frenzy:
```

Comment:
It looks like Cow's `AttitudeToPlayer` is often `ATTITUDE_Frenzy` if the player attacks the cow, so even though the cow was innocent, the player gets a score for killing it.

Reply:
I make this code after analyze few evil cows. So there dilemmas: or you count score for some kill good cows, or you make negative score for some kill evil cows. Or such cows kill you.
```
class bcCow expands Cow;

defaultproperties
{
				Aggressiveness=1.500000
				bIsWuss=False
				JumpZ=250.000000
				MaxStepHeight=200.000000
				FovAngle=120.000000
				Health=250
				AttitudeToPlayer=ATTITUDE_Frenzy
				Intelligence=BRAINS_MAMMAL
				CombatStyle=1.000000
				MenuName="Evil Cow"
				Mass=12000.000000
}
```

Reply:
Odd, this commit makes it seem like `NaliRabbit` is not accounted for anymore.

Is it because it's easy to accidentally stomp one? :p

Reply:
It's because `NaliRabit` is unfortunately not a `ScriptedPawn` (incorrect assumption by me originally). The rest of the code here is in a `ScriptedPawn`-only block.

Reply:
Ahhh, okay. I see. Cool!